v1.7 backports 2020-05-20

jrajahalme opened the pull request:
* #11452 -- cilium: rename aws/eni/routing to pkg/datapath/linux/routing (@bpineau)
 * #11541 -- envoy: Take xds mutator lock for map access (@jrajahalme)
 * #11555 -- Shorten dummy device name in linuxrouting tests (@christarazi)
 * #11530 -- istio: Update to 1.5.4 (@jrajahalme)
 * #11599 -- Fix pre-flight deployment (@aanm) (last commit modifying .github/workflows not backported)
 * #11566 -- cilium: Add CLI to introspect IP <-> Identity cache (@joestringer)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 11452 11541 11555 11530 11599 11566; do contrib/backporting/set-labels.py $pr done 1.7; done
```

joestringer commented on pkg/endpoint/endpoint.go:
```diff
@@ -2097,18 +2097,18 @@ func (e *Endpoint) Delete(monitor monitorOwner, ipam ipReleaser, manager endpoin
 	}
 
-	if option.Config.IPAM == option.IPAMENI {
+	if option.Config.IPAM == option.IPAMENI || option.Config.IPAM == option.IPAMAzure {
```
v1.7 doesn't have Azure IPAM, did this backport cleanly?

christarazi replied:
I marked the original PR as backport. But I wasn't aware we don't have Azure IPAM in v1.7. That PR title doesn't make it clear that it's adding support for Azure.

A potential thing we can try is to remove lines like these, but in general keep the refactoring changes that that PR brings.

joestringer replied:
I thought it might be a concern when we merged it, but figured that you would backport it given backports duty last week. I suspect I didn't actually voice this concern anywhere though :-)

We should likely drop all azure references from this backport commit.

christarazi replied:
@jrajahalme There are still references to Azure IPAM in the code. Could you please remove those? All the references come from https://github.com/cilium/cilium/pull/11452

jrajahalme replied:
Done. Someone had already reverted the line above, but there were some logs and comments that I reverted back to original.

christarazi reviewed (approved):
My changes LGTM

christarazi commented on pkg/datapath/linux/routing/info.go:
```diff
@@ -23,22 +23,22 @@ import (
-// connectivity via the local rule and route tables while in ENI mode. The
-// information in this struct is used to create rules and routes which direct
-// traffic out of the ENI devices (egress).
+// connectivity via the local rule and route tables while in ENI or Azure IPAM mode.
```
https://github.com/cilium/cilium/pull/11630#discussion_r430594899

jrajahalme replied:
reverted

christarazi commented on pkg/datapath/linux/routing/routing.go:
```diff
@@ -30,16 +30,17 @@ import (
-// Configure sets up the rules and routes needed when running in ENI mode.
-// These rules and routes direct egress traffic out of the ENI device and
+// Configure sets up the rules and routes needed when running in ENI or
+// Azure IPAM mode.
```
https://github.com/cilium/cilium/pull/11630#discussion_r430594899

jrajahalme replied:
reverted

christarazi commented on daemon/daemon.go:
```diff
@@ -139,8 +139,8 @@ type Daemon struct {
 	// healthEndpointRouting is the information required to set up the health
-	// endpoint's routing in ENI mode
-	healthEndpointRouting *enirouting.RoutingInfo
+	// endpoint's routing in ENI or Azure IPAM mode
```
https://github.com/cilium/cilium/pull/11630#discussion_r430594899

jrajahalme replied:
reverted

christarazi reviewed (changes requested):
I've pointed out the Azure references per https://github.com/cilium/cilium/pull/11630#discussion_r430594899

bpineau committed:
cilium: gc routes on endpoint deletion on Azure

[ upstream commit fac18020d6724427c4b0ac36e72d4dcee04f15fe ]

Also garbage collect routes on endpoint deletion when running
Azure IPAM mode (the same way we do on AWS instances).

And rename aws/eni/routing to pkg/datapath/linux/routing:
per discussion in #11268 : the routing rules we use here are in
no way specific to AWS ENI; they are for also used to setup
Azure IPAM routes.

Signed-off-by: Benjamin Pineau <benjamin.pineau@datadoghq.com>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
envoy: Take xds mutator lock for map access

[ upstream commit 003efe47c0662659243ce68d92c85c1f5265b403 ]

'ackedVersions' was accessed in DeleteNode() without taking the mutex
which can lead to mutating the map while readers are accessing it.

Fixes: #11535
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

christarazi committed:
Shorten dummy device name in linuxrouting tests

[ upstream commit 2141ecac532028b885f10d927ee590de63e3eb38 ]

Due to the refactor from #11452, the device name was renamed from
"enirouting-test" to "linuxrouting-test" which is 17 characters long,
exceeding the max device name size (`IFNAMSIZ`) of 16. [1]

This commit renames the device name to be under 16 characters.

Fixes #11547
Fixes #11452

[1]:
https://elixir.bootlin.com/linux/v5.6/source/include/uapi/linux/if.h#L33

Signed-off-by: Chris Tarazi <chris@isovalent.com>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

aanm committed:
preflight/templates: do not set 'env' if unnecessary

[ upstream commit 2d5333f5361cd39fc8b453207ffe28b01a517acc ]

The 'env' section can be set even if no environments are being set by
the user. With this commit we only set the 'env' section if one of the
environment variables are set by the user.

Fixes: ff863a8f20b2 ("add validation checker for installed CNPs in preflight")
Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
istio: Update to 1.5.4

[ upstream commit eaea1369e66d5227ec326a929199db5e049f1555 ]

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

aanm committed:
cilium/cmd: do not fail if CCNPs are not found

[ upstream commit be8b9431ac4190d43ee3ee0c2055d6d45805e7ad ]

In a cluster being upgraded from < 1.7, it is expected that CCNPs are
not available. For this reason, the CNP validation tool should not fail
has it is expected for CCNPs to not be available.

Fixes: ff863a8f20b2 ("add validation checker for installed CNPs in preflight")
Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

joestringer committed:
cilium: Add CLI to introspect IPIdentityCache

[ upstream commit f8296364434774e435d3173c36443ea630c8c6d5 ]

The flow of IP <-> Identity mappings into the IPIdentityCache follow the
path of:

1) K8s watch / local poll / generation of the mapping
2) pkg/ipcache:IPIdentityCache
3) pkg/bpf/map userspace cache of map entries being pushed into BPF map
4) Actual BPF map

We can currently introspect some of (1) depending on the source; (3)
always; and (4) depending on the kernel version in use. (2) cannot be
introspected.

This commit adds support for introspecting (2) to assist in
troubleshooting issues related to the propagation of IP to Identity
mappings.

Example output:

    # cilium ip list
    IP                 IDENTITY   SOURCE
    0.0.0.0/0          2
    10.29.2.92/32      51185      k8s
    10.29.25.191/32    1
    10.29.27.15/32     4
    10.29.27.93/32     104        k8s
    10.29.74.136/32    30909      k8s
    10.29.78.12/32     16083      k8s
    10.29.89.252/32    20623      k8s
    10.29.136.128/32   3157       k8s
    10.29.177.163/32   19933      k8s
    10.29.194.215/32   22010      k8s
    10.29.209.106/32   59801      k8s
    10.29.224.16/32    589        k8s
    192.168.1.29/32    1

Signed-off-by: Joe Stringer <joe@cilium.io>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

aanm committed:
docs: add missing --namespace in upgrade guide

[ upstream commit b8aa5c7d2b717d79c1cc4a198344d24a203adb50 ]

Fixes: f611334d4b7d ("Use helm repository in docs")
Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

aanm committed:
preflight/templates: add clusterrole for pre-flight

[ upstream commit 5c950b16fda091d18b890a87544165fb3b646cc5 ]

The pre-flight deployment and daemonset might have different requirements
for the clusterrole deployed in the cluster. Adding a specific clusterrole
for the pre-flight allows users to run it successfully.

Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
